packages/calcite-components/src/components/input-date-picker/input-date-picker.tsx

```diff
@@ -130,11 +121,25 @@ export class InputDatePicker
 
   private focusOnOpen = false;
 
-  focusTrap: FocusTrap;
-
-  private focusTrapDeactivates = (): void => {
-    this.open = false;
-  };
+  focusTrap = useFocusTrap<this>({
+    triggerProp: "open",
+    focusTrapOptions: {
+      onActivate: () => {
+        if (this.focusOnOpen) {
+          this.datePickerEl?.setFocus();
+          this.focusOnOpen = false;
+        }
+      },
+      allowOutsideClick: true,
+      // Allow outside click and let the popover manager take care of closing the popover.
+      clickOutsideDeactivates: false,
```

Comment:
Nitpick: Do we need to specify this? 

`clickOutsideDeactivates: false` is the default option.

Reply:
It's false in `focus-trap`, but we override the default in our [controller/utils](https://github.com/Esri/calcite-design-system/blob/dev/packages/calcite-components/src/utils/focusTrapComponent.ts#L64).

Reply:
Side note: I'll get rid of the comment since it's not applicable.

Reply:
> It's false in `focus-trap`, but we override the default in our [controller/utils](https://github.com/Esri/calcite-design-system/blob/dev/packages/calcite-components/src/utils/focusTrapComponent.ts#L64).

Interesting, is this intentional or do we need to specify it at component level?

Reply:
I believe it's intentional. We can revisit it if needed. 😄